vpc/service/service_test.go

```diff
@@ -55,16 +54,22 @@ func TestService(t *testing.T) {
 			BranchNetworkInterfaceDescription: vpc.DefaultBranchNetworkInterfaceDescription,
 
 			disableRouteCache: true,
-		})
+		}
+		vpcService, err := NewVpcService(ctx, vpcServiceConfig)
+		if err != nil {
+			panic(err)
+		}
+		serverErr := vpcService.Run(ctx, addr)
 		if serverErr != nil && serverErr != context.Canceled {
 			panic(serverErr)
 		}
 	}()
+	time.Sleep(time.Second * 2)
```

Comment:
why do we need a sleep here?

Reply:
In the past, the caller of `Run` function starts listening and then pass the listener to `Run`. With this refactoring, the listening is started inside `Run` and the caller only passes an address. In this particular test, before the refactoring, listening is done in the main thread. After the refactoring, listening is done in the go routine where `Run` is called. So it may take a while before the health check starts. Actually, even in the old code, if the whole RPC server setup is done in a separate go routine, then it may take a while before the health check in the main thread to pass. I guess moving listening to `Run` just made this time difference larger.

Sleep is not ideal. Ideally, we should have a retry with timeout. I'll do that in later cleanup.